db/lastfm_session.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+import os
+import binascii
+import db
+from sqlalchemy import text
+from lastfm_user import User
+
+
+class Session(object):
+    """ Session class required by the api-compat """
+
+    def __init__(self, row):
+        self.id, userid, self.sid, self.api_key, self.timestamp = row
+        self.user = User.load_by_id(userid)
+
+    @staticmethod
+    def load(session, api_key=None):
+        """ Load the session details from the database.
+            If the api_key is also supplied then verify it as well.
+        """
+        params = {'sid': session}
+        query = "SELECT * FROM session WHERE sid=:sid"
+        if api_key:
+            params['api_key'] = api_key
+            query = "SELECT * FROM session WHERE sid=:sid AND api_key=:api_key"
+
+        with db.engine.connect() as connection:
+            result = connection.execute(text(query), params)
```

Comment:
This comment applies to the whole codebase -- please review the entire codebase for instances where this comment might apply.

Anywhere you make calls to the database you need to catch DB exceptions and handle them appropriately.  I would suggest that in the API handlers that you catch the exceptions there and then raise the appropriate HTTP exception. This way the user gets more informative information than just a 500 error that may not make sense to them.